pypeit/core/wavecal/autoid.py

```diff
@@ -688,6 +695,129 @@ def reidentify(spec, spec_arxiv_in, wave_soln_arxiv_in, line_list, nreid_min, de
+        nreid_min (int): 
+            Minimum number of times that a given candidate reidentified line must be properly matched with a line in the arxiv
+            to be considered a good reidentification. If there is a lot of duplication in the arxiv of the spectra in question
+            (i.e. multislit) set this to a number like 2-4. For echelle this depends on the number of solutions in the arxiv.
+            For fixed format echelle (ESI, X-SHOOTER, NIRES) set this 1. For an echelle with a tiltable grating, it will depend
+            on the number of solutions in the arxiv.
+        match_toler (float, optional): Defaults to 2.0.
+            Matching tolerance in pixels for a line reidentification. A good line match must match within this tolerance to the
+            the shifted and stretched archive spectrum, and the archive wavelength solution at this match must be within
+            match_toler dispersion elements from the line in line list.
+        nonlinear_counts (float, optional): Defaults to 1e10.
+            For arc line detection: Arc lines above this saturation
+            threshold are not used in wavelength solution fits because
+            they cannot be accurately centroided
+        sigdetect (float, optional): Defaults to 5.0.
+            Threshold for detecting arcliens
+        fwhm (float, optional): Defaults to 4.0.
+            Full width at half maximum for the arc lines
+        debug_peaks (bool, optional): Defaults to False.
+        use_unknowns (bool, optional): Defaults to False.
+            If True, use the unknowns in the solution (not recommended)
+
+    Returns:
+        tuple: tcent (np.ndarray; centroid of lines), spec_cont_sub (np.ndarray; subtracted continuum), 
+            patt_dict_slit (dict; dictionary on the lines), tot_line_list (astropy.table.Table; line list)
+    """
+    # Load line list
+    tot_line_list, _, _ = waveio.load_line_lists(lamps, include_unknown=use_unknowns)
+
+
+    # Generate the wavelengths from the line list and sort
+    wvdata = np.array(tot_line_list['wave'].data)  # Removes mask if any
+    wvdata.sort()
+
+    # Search for lines in the input arc
+    tcent, ecent, cut_tcent, icut, spec_cont_sub = wvutils.arc_lines_from_spec(
+        spec, sigdetect=sigdetect,
+        nonlinear_counts=nonlinear_counts, 
+        fwhm=fwhm, debug = debug_peaks)
+    # If there are no lines in the input arc, return
+    if tcent.size == 0:
+        return None
+
+    # Search for lines in the arxiv arc
+    tcent_arxiv, ecent_arxiv, cut_tcent_arxiv, icut_arxiv, spec_cont_sub_now = wvutils.arc_lines_from_spec(
+            spec_arxiv, sigdetect=sigdetect,
+            nonlinear_counts=nonlinear_counts, fwhm = fwhm, debug = debug_peaks)
+    # If there are no lines in the arxiv arc, return
+    if tcent_arxiv.size == 0:
+        return None
+
+    # Interpolate the input wavelengths
+    fwv_guess = scipy.interpolate.interp1d(np.arange(len(wv_guess)), wv_guess, 
```

Comment:
Given how much we use the `interp1d` function in PypeIt, I find it concerning that SciPy are considering deleting the `interp1d` function in future releases... a problem for the future  :-)